File: python/002/main.py
# ProjectEuler 002
# Python / main.py
# copyright (c) 2015 Susisu
#
# Python 3.4.2
# `python3 main.py`


def main():
    sum = 0
    a = 0
    b = 1
    while a <= 4000000:
        a, b = b, a + b
        if a % 2 == 0:
            sum += a
    print(sum)

main()

File: ruby/002/main.rb
=begin
    ProjectEuler 002
    Ruby / main.rb
    copyright (c) 2015 Susisu

    Ruby 2.0.0p481
    `ruby main.rb`
=end

def main
    sum = 0
    a = 0
    b = 1
    while a <= 4000000 do
        a, b = b, a + b
        if a % 2 == 0 then
            sum += a
        end
    end
    puts sum
end

main()
